src/prepare.rs

```diff
@@ -3,12 +3,77 @@
 use crate::*;
 pub use std::fmt::Write;
 
-#[derive(Debug, Default)]
-pub struct SqlWriter {
-    pub(crate) counter: usize,
+pub trait SqlWriter: Write {
+    fn push_param(&mut self, value: Value, query_builder: &dyn QueryBuilder);
+    fn result(self) -> String;
+}
```

Comment:
Hey @ikrivosheev, I think we don't need the `result()` method. The `SqlWriter` trait is only responsible for writing SQL and push parameters.

The `to_string` alike method can be place in struct `SqlStringWriter` and `SqlWriterObj`.

Reply:
@billy1624 hello! 

1. `trait SqlWriter: Writer + ToString`
2. remove `SqlStringWriter`, use `String` and `impl SqlWriter for String`
3. Rename `SqlWriterObj` to `SqlWriterValues`